platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/ArchUnitTests.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package platform.tooling.support.tests;
+
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+
+import org.junit.jupiter.api.Test;
+import platform.tooling.support.Helper;
+
+class ArchUnitTests {
+
+	@Test
+	void acyclicImportJars() {
+		var jarFiles = Helper.loadJarFiles();
+		var classes = new ClassFileImporter().importJars(jarFiles);
+		// about 928 classes found in all jars
+		assertTrue(classes.size() > 800, "expected more than 800 classes, got: " + classes.size());
+
+		assertAcyclic(classes);
+
+		// LoggerFactory.java:80 -> sets field LoggerFactory$DelegatingLogger.julLogger
+		// GeneralCodingRules.NO_CLASSES_SHOULD_USE_JAVA_UTIL_LOGGING.check(classes);
+
+		// LoggerFactory.java:155 -> new Throwable()
+		// GeneralCodingRules.NO_CLASSES_SHOULD_THROW_GENERIC_EXCEPTIONS.check(classes);
```

Comment:
@codecholeric How do I filter out a single class from a JavaClass instance? `org.junit.platform.commons.logging.LoggerFactory` should be allowed to break both generic rules.

Reply:
You can always filter the classes by

```
JavaClasses subset = classes.that(predicate);
```

In your case it could be

```
import static com.tngtech.archunit.base.DescribedPredicate.not;
import static com.tngtech.archunit.lang.conditions.ArchPredicates.are;
import static com.tngtech.archunit.core.domain.JavaClass.Predicates.equivalentTo;

classes.that(are(not(equivalentTo(LoggerFactory.class)))
```